modules/Bio/EnsEMBL/BaseAlignFeature.pm

```diff
@@ -1009,7 +1012,303 @@ sub _query_unit {
   throw( "Abstract method call!" );
 }
 
+=head2 _mdtag_alignment_length
+
+  Arg [1]    : None
+  Description: return the alignment length (including indels) based on the mdtag (mdz) string
+  Returntype : int
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _mdtag_alignment_length {
+  my $self = shift;
+
+  if (! defined $self->{'_alignment_length'} && defined $self->cigar_string) {
+    my $mdz_string =  $self->cigar_string;
+    my $chunks = $self->_get_mdz_chunks($mdz_string);
+    my $alignment_length = 0;
+    $alignment_length = $self->_get_mdz_alignment_length($chunks);
+    $self->{'_alignment_length'} = $alignment_length;
+  }
+  return $self->{'_alignment_length'};
+}
+
+=head2 _get_mdz_chunks
+
+  Arg [1]    : mdz string
+  Description: parses the mdz string and group it according the type
+  Returntype : array of strings
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _get_mdz_chunks {
+  my $self = shift;
+
+  my $mdz_string = shift;
+  my $alignment_length = 0;
+  my @chunks;
+    if ( $mdz_string =~ /^MD:Z:(.*)/ ){
```

Comment:
Perhaps that should be ^MD:Z:(.+), because with the * "MD:Z:" would match.

Reply:
This issue still exists.

Reply:
I agree, this should be addressed.